runtime/jvmti/jvmtiHelpers.c

```diff
@@ -403,6 +387,20 @@ allocateEnvironment(J9InvocationJavaVM * invocationJavaVM, jint version, void **
 UDATA
 prepareForEvent(J9JVMTIEnv * j9env, J9VMThread * currentThread, J9VMThread * eventThread, UDATA eventNumber, jthread * threadRefPtr, UDATA * hadVMAccessPtr, UDATA wantVMAccess, UDATA jniRefSlots, UDATA * javaOffloadOldState)
 {
+	J9JVMTIThreadData *threadData = NULL;
+#if JAVA_SPEC_VERSION >= 19
+	j9object_t threadObj = currentThread->threadObject;
+	/* threadObj can be null early in J9VMThread initialization. */
+	if (NULL != threadObj) {
+		void *tlsArray = J9OBJECT_ADDRESS_LOAD(currentThread, threadObj, currentThread->javaVM->tlsOffset);
+		if (NULL != tlsArray) {
+			threadData = jvmtiTLSGet(currentThread, threadObj, j9env->tlsKey);
+		}
+	}
+#else
+	threadData = jvmtiTLSGet(currentThread, currentThread->threadObject, j9env->tlsKey);
```

Comment:
Should there not be a NULL check on the tlsArray as above?

Reply:
For java 18 and below we are continuing to use the os thread's tls array, which won't be null.